hera_cal/lstbin_simple.py

```diff
@@ -0,0 +1,1677 @@
+    sigma_clip_thresh
+        The number of standard deviations to use as a threshold for sigma clipping.
+        If 0, no sigma clipping is performed. Note that sigma-clipping is performed
+        per baseline, frequency, and polarization.
+    sigma_clip_min_N
+        The minimum number of unflagged samples required to perform sigma clipping.
+
+    Returns
+    -------
+    out_data, out_flags, out_std, out_nsamples
+        The reduced data, flags, standard deviation (across days) and nsamples.
+        Shape ``(nbl, nfreq, npol)``.
+    """
+    # data has shape (ntimes, nbl, npols, nfreqs)
+    # all data is assumed to be in the same LST bin.
+
+    assert data.shape == nsamples.shape == flags.shape
+    
+    if not mutable:
+        flags = flags.copy()
+        nsamples = nsamples.copy()
+        data = data.copy()
+
+    flags[np.isnan(data) | np.isinf(data) | (nsamples == 0)] = True
+
+    # Flag entire LST bins if there are too many flags over time
+    flag_frac = np.sum(flags, axis=0) / flags.shape[0]
+    nflags = np.sum(flags)
+    logger.info(f"Percent of data flagged before thresholding: {100*np.sum(flags)/flags.size:.2f}%")
+    flags |= flag_frac > flag_thresh
+    data[flags] *= np.nan  # do this so that we can do nansum later. multiply to get both real/imag as nan
+    logger.info(f"Flagged a further {100*(np.sum(flags) - nflags)/flags.size:.2f}% of visibilities due to flag_frac > {flag_thresh}")
+
+    # Now do sigma-clipping.
+    if sigma_clip_thresh > 0:
+        nflags = np.sum(flags)
+        flags |= sigma_clip(data, sigma=sigma_clip_thresh, min_N = sigma_clip_min_N)
+        data[flags] *= np.nan
+        logger.info(f"Flagged a further {100*(np.sum(flags) - nflags)/flags.size:.2f}% of visibilities due to sigma clipping")
+
+    # get other stats
+    logger.info("Calculating std")
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", message="Degrees of freedom <= 0 for slice.")
+        std = np.nanstd(data.real, axis=0) + 1j*np.nanstd(data.imag, axis=0)
+        
+    nsamples[flags] = 0
+    norm = np.sum(nsamples, axis=0)  # missing a "clip" between 1e-99 and inf here...
+    
+    logger.info("Calculating mean")
+    data = np.nansum(data * nsamples, axis=0)
+    data[norm>0] /= norm[norm>0]
+    data[norm<=0] = 1  # any value, it's flagged anyway
+        
+    f_min = np.all(flags, axis=0)
+    std[f_min] = 1.0
+    norm[f_min] = 0  # This is probably redundant.
+
+    return data, f_min, std, norm
+
+def adjust_lst_bin_edges(lst_bin_edges: np.ndarray) -> np.ndarray:
+    """
+    Adjust the LST bin edges so that they start in the range [0, 2pi) and increase.
+    
+    Performs the adjustment in-place.
+    """
+    while lst_bin_edges[0] < 0:
+        lst_bin_edges += 2*np.pi
+    while lst_bin_edges[0] >= 2*np.pi:
+        lst_bin_edges -= 2*np.pi
+
+def lst_bin_files_for_baselines(
+    data_files: list[Path | FastUVH5Meta], 
+    lst_bin_edges: np.ndarray, 
```

Comment:
same comment about bin edges here

Reply:
I've updated the docstring to match the one for `lst_align` based on your previous comment. I still think that edges are better, but I guess I could be persuaded.